fs/accounting.go

```diff
@@ -21,20 +21,70 @@ var (
 	tokenBucketMu   sync.Mutex // protects the token bucket variables
 	tokenBucket     *tb.Bucket
 	prevTokenBucket = tokenBucket
+	currLimitMu     sync.Mutex // protects changes to the timeslot
+	currLimit       BwTimeSlot
 )
 
 // Start the token bucket if necessary
 func startTokenBucket() {
-	if bwLimit > 0 {
-		tokenBucket = tb.NewBucket(int64(bwLimit), 100*time.Millisecond)
-		Log(nil, "Starting bandwidth limiter at %vBytes/s", &bwLimit)
+	currLimitMu.Lock()
+	currLimit, err := bwLimit.LimitAt(time.Now())
+	currLimitMu.Unlock()
+	// This should never happen! It means we got an invalid time from time.Now()
+	if err != nil {
+		Log(nil, "This error should not happen. Please open a bug: %v", err)
+		return
+	}
+
+	if currLimit.bandwidth > 0 {
+		tokenBucket = tb.NewBucket(int64(currLimit.bandwidth), 100*time.Millisecond)
+		Log(nil, "Starting bandwidth limiter at %vBytes/s", &currLimit.bandwidth)
 
 		// Start the SIGUSR2 signal handler to toggle bandwidth.
 		// This function does nothing in windows systems.
 		startSignalHandler()
 	}
 }
 
+// startTokenTicker creates a ticker to update the bandwidth limiter every minute.
+func startTokenTicker() {
+	ticker := time.NewTicker(time.Minute)
+	go func() {
+		for _ = range ticker.C {
+			limitNow, err := bwLimit.LimitAt(time.Now())
+			if err != nil {
+				Log(nil, "Unexpected error reading bandwidth limit: %v", err)
+				continue
+			}
+			currLimitMu.Lock()
+
+			if currLimit.bandwidth != limitNow.bandwidth {
+				tokenBucketMu.Lock()
+				if tokenBucket != nil {
+					err := tokenBucket.Close()
+					if err != nil {
+						Log(nil, "Error closing token bucket: %v", err)
+						continue
```

Comment:
The idea was that if Close() fails, we don't attempt any operations on tokenBucket, but I'm guessing here that Close() should be a fire & forget operation anyway, since it's used in defer. Removed the "continue".